packages/react/src/date-picker/utils.ts

```diff
@@ -90,9 +91,16 @@ export function transformValuePropToInputValue(
 	valueProp: Date | string | undefined,
 	dateFormat: AcceptedDateFormats
 ): string {
-	if (typeof valueProp === 'string') return valueProp;
+	if (
+		typeof valueProp === 'string' &&
+		parseISO(valueProp).toString() === 'Invalid Date'
+	)
+		return valueProp;
 	if (typeof valueProp === 'undefined') return '';
-	if (isValidDate(valueProp)) return formatDate(valueProp, dateFormat);
+
+	const valueAsDate =
```

Comment:
Should this also be using `normaliseDateString`? Isn't that how we can guarantee that the value is a date?